Template10 (Library)/Controls/HamburgerMenu.xaml.cs

```diff
@@ -513,64 +515,81 @@ public void UpdatePaneMarginToShowHamburgerButton()
-            if (AllNavButtonsAreLoaded && value?.PageType != null)
+            if (AllNavButtonsAreLoaded && current?.PageType != null)
             {
-                if (await NavigationService.NavigateAsync(value.PageType, value?.PageParameter, value?.NavigationTransitionInfo))
+                if (await NavigationService.NavigateAsync(current.PageType, current?.PageParameter, current?.NavigationTransitionInfo))
                 {
+                    SignalPreviousPage(previous, current);
+                    if (current != null) SignalCurrentPage(previous, current);
+
                     IsOpen = (DisplayMode == SplitViewDisplayMode.CompactInline && IsOpen);
-                    if (value.ClearHistory)
+                    if (current.ClearHistory)
                         NavigationService.ClearHistory();
-                    if (value.ClearCache)
+                    if (current.ClearCache)
                         NavigationService.ClearCache(true);
                 }
-                else if (NavigationService.CurrentPageType == value.PageType
-                     && (NavigationService.CurrentPageParam ?? string.Empty) == (value.PageParameter ?? string.Empty))
+                else if (NavigationService.CurrentPageType == current.PageType && (NavigationService.CurrentPageParam ?? string.Empty) == (current.PageParameter ?? string.Empty))
                 {
-                    if (value.ClearHistory)
+                    SignalPreviousPage(previous, current);
+                    if (current != null) SignalCurrentPage(previous, current);
+
+                    if (current.ClearHistory)
                         NavigationService.ClearHistory();
-                    if (value.ClearCache)
+                    if (current.ClearCache)
                         NavigationService.ClearCache(true);
                 }
-                else if (NavigationService.CurrentPageType == value.PageType)
+                else if (previous == null || NavigationService.CurrentPageType == current.PageType)
                 {
-                    // just check it
+                    if (current != null) SignalCurrentPage(previous, current);
                 }
                 else
                 {
+                    // Re-instate Selected to previous page, but avoid calling this method (UpdateSelectedAsync) all over
+                    // again, and we use a flag to effect this. See InternalSelectedChanged() method where it's used.
+
+                    IsCanceledNavigation = true;
+                    Selected = previous;
+                    IsCanceledNavigation = false;
+                    current.IsChecked = false;
+                    current.RaiseUnselected();
                     return;
                 }
             }
+            else
+            {
+                SignalPreviousPage(previous, current);
+                if (current != null) SignalCurrentPage(previous, current);
```

Comment:
Don't you think (current != null) belongs in SignalCurrentPage() instead of here?